## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 455551`
Commit: [`e38da2b7a7d35fb1d8e64894369808fbc790e6cf`](https://github.com/NixOS/nixpkgs/commit/e38da2b7a7d35fb1d8e64894369808fbc790e6cf) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/e38da2b7a7d35fb1d8e64894369808fbc790e6cf..pull/455551/head))
Merge: [`e6b31429698cbf549adab00c1e385cf031230f45`](https://github.com/NixOS/nixpkgs/commit/e6b31429698cbf549adab00c1e385cf031230f45)

Logs: https://github.com/yzx9/nixpkgs-review-gha/actions/runs/18804358587


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>djv</li>
    <li>feather-tk</li>
    <li>nativefiledialog-extended</li>
    <li>nativefiledialog-extended.dev</li>
    <li>tlrender</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>djv</li>
    <li>feather-tk</li>
    <li>nativefiledialog-extended</li>
    <li>nativefiledialog-extended.dev</li>
    <li>tlrender</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>djv</li>
    <li>feather-tk</li>
    <li>tlrender</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nativefiledialog-extended</li>
    <li>nativefiledialog-extended.dev</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>feather-tk</summary>
<pre>  286 | FSFindFolder(
      | ^
/nix/build/nix-10454-987098220/source/lib/feather-tk/core/FileUnix.cpp:38:9: warning: &#x27;FSRefMakePath&#x27; is deprecated: first deprecated in macOS 10.8 [-Wdeprecated-declarations]
   38 |         FSRefMakePath(&amp;ref, (UInt8*)&amp;path, PATH_MAX);
      |         ^
/nix/store/4mn3pvf8ahm21qjdh4r2w1lf90gcv9yq-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/Files.h:4086:18: note: &#x27;FSRefMakePath&#x27; has been explicitly marked deprecated here
 4086 | extern OSStatus  FSRefMakePath(const FSRef *ref, UInt8 *path, UInt32 pathBufferSize)     __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
      |                  ^
2 warnings generated.
[ 60%] Linking CXX shared library libfeather-tk-core.dylib
Undefined symbols for architecture x86_64:
  &quot;_FSFindFolder&quot;, referenced from:
      feather_tk::getUserPath(feather_tk::UserPath) in FileUnix.cpp.o
  &quot;_FSRefMakePath&quot;, referenced from:
      feather_tk::getUserPath(feather_tk::UserPath) in FileUnix.cpp.o
ld: symbol(s) not found for architecture x86_64
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/feather-tk/core/CMakeFiles/feather-tk-core.dir/build.make:728: lib/feather-tk/core/libfeather-tk-core.dylib] Error 1
make[1]: *** [CMakeFiles/Makefile2:304: lib/feather-tk/core/CMakeFiles/feather-tk-core.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nativefiledialog-extended</li>
    <li>nativefiledialog-extended.dev</li>
  </ul>
</details>

--- a/pkgs/by-name/fe/feather-tk/package.nix
+++ b/pkgs/by-name/fe/feather-tk/package.nix
@@ -21,6 +21,7 @@
   enablePython ? false,
   enableTests ? false,
   enableExamples ? false,
+  enableShared ? !stdenv.hostPlatform.isStatic,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -31,7 +32,6 @@ stdenv.mkDerivation (finalAttrs: {
     owner = "darbyjohnston";
     repo = "feather-tk";
     tag = finalAttrs.version;
-    fetchSubmodules = true;
     hash = "sha256-hcV99y14o3YFUtKDLEKaR7MxBB3pBdd3sferrYvtvYw=";
   };
 
@@ -63,6 +63,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   cmakeFlags = [
     (lib.cmakeFeature "CMAKE_BUILD_TYPE" "Release")
+    (lib.cmakeBool "BUILD_SHARED_LIBS" enableShared)
     (lib.cmakeBool "feather_tk_UI_LIB" true)
     (lib.cmakeFeature "feather_tk_API" "GL_4_1")
     (lib.cmakeBool "feather_tk_nfd" enableNFD)

--- a/pkgs/by-name/fe/feather-tk/package.nix
+++ b/pkgs/by-name/fe/feather-tk/package.nix
@@ -11,13 +11,13 @@
   libGL,
   libpng,
   lunasvg,
+  nativefiledialog-extended,
   nlohmann_json,
   plutovg,
   xorg,
   zlib,
-  nativeFileDialog ? null,
   python3Packages ? null,
-  enableNFD ? false,
+  enableNFD ? true,
   enablePython ? false,
   enableTests ? false,
   enableExamples ? false,
@@ -51,8 +51,8 @@ stdenv.mkDerivation (finalAttrs: {
     zlib
     libGL
   ]
-  ++ lib.optionals (enableNFD && nativeFileDialog != null) [
-    nativeFileDialog
+  ++ lib.optionals enableNFD [
+    nativefiledialog-extended
   ]
   ++ lib.optionals (enableNFD && stdenv.isLinux) [
     gtk3

--- a/pkgs/by-name/na/nativefiledialog-extended/package.nix
+++ b/pkgs/by-name/na/nativefiledialog-extended/package.nix
@@ -1,0 +1,75 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  SDL2,
+  SDL2_ttf,
+  gtk3,
+  testers,
+  enableShared ? !stdenv.hostPlatform.isStatic,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "nativefiledialog-extended";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "btzy";
+    repo = "nativefiledialog-extended";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-GwT42lMZAAKSJpUJE6MYOpSLKUD5o9nSe9lcsoeXgJY=";
+  };
+
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    SDL2
+    SDL2_ttf
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isLinux [
+    gtk3
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "BUILD_SHARED_LIBS" enableShared)
+    (lib.cmakeBool "nfd_ROOT_PROJECT" true)
+    (lib.cmakeBool "NFD_BUILD_TESTS" true)
+    (lib.cmakeBool "NFD_BUILD_SDL2_TESTS" true)
+    (lib.cmakeBool "NFD_INSTALL" true)
+  ];
+
+  # NOTE: Although the tests have been compiled, they still require GUI
+  # interactions. Let's disable running them for now.
+  doCheck = false;
+
+  passthru.tests = {
+    cmake-config = testers.hasCmakeConfigModules {
+      package = finalAttrs.finalPackage;
+      moduleNames = [ "nfd" ];
+    };
+  };
+
+  meta = {
+    homepage = "https://github.com/btzy/nativefiledialog-extended";
+    description = "Cross platform native file dialog library with C and C++ bindings";
+    longDescription = ''
+      A small C library with that portably invokes native file open, folder
+      select and file save dialogs. Write dialog code once and have it pop up
+      native dialogs on all supported platforms. Avoid linking large
+      dependencies like wxWidgets and Qt.
+    '';
+    license = lib.licenses.zlib;
+    maintainers = with lib.maintainers; [ yzx9 ];
+    platforms = with lib.platforms; linux ++ darwin;
+  };
+})

